src/Uno.UI/DataBinding/BindingPath.cs

```diff
@@ -696,14 +720,15 @@ private void BuildValueSetter()
 			{
 				if (_valueSetter == null && _dataContextType != null)
 				{
+					_valueSetter = BindingPropertyHelper.GetValueSetter(
+						_dataContextType,
+						PropertyName,
+						convert: true,
+						precedence: _precedence ?? DependencyPropertyValuePrecedences.Local
+					);
 					if (_precedence == null)
 					{
-						BuildLocalValueSetter();
-						_valueSetter = _localValueSetter;
-					}
-					else
-					{
-						_valueSetter = BindingPropertyHelper.GetValueSetter(_dataContextType, PropertyName, convert: true, precedence: _precedence.Value);
+						_localValueSetter = _valueSetter;
```

Comment:
I don't see the point to remove this optimization? If I read it correctly, here if the `_precedence` is null, you will build a second `_valueSetter` which is the same as the `_localValueSetter`

Reply:
we will create a _valueSetter for the specified precedence in any case
// now if the precedence is unspecified, we default to the `local` precedence
so if unspecified, _valueSetter is also good for _localValueSetter